src/common/http/types.h

```diff
@@ -14,6 +15,41 @@ enum Method {
     UnknownMethod,
 };
 
+constexpr const char *to_str(Method method) {
+    switch (method) {
+    case Head:
+        return "HEAD";
+    case Get:
+        return "GET";
+    case Post:
+        return "POST";
+    case Put:
+        return "PUT";
+    case Delete:
+        return "DELETE";
+    default:
+        assert(0);
+    case UnknownMethod:
+        return "UNKNOWNMETHOD";
```

Comment:
The reason I chose to not `assert` the `UnknownMethod` the `to_str` one is because the general use case for this method I have in my head is logging. So when a user decides to output log an unknown state (let's say during an initialization of some object somewhere), the user might be confused why a simple 'log' would crash their program. However in the second case, calling it with  `UnknownMethod` feels like 'always a bug'.